tests/synthdnn/synthdnn.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2025 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import argparse
+import os
+from tempfile import NamedTemporaryFile
+
+from matmul import sampler as matmul_sampler
+from matmul import primitive as matmul
+
+
+def log(output):
+    print("synthdnn: " + output)
+
+def error(output):
+    print("synthdnn: error: " + output)
+    exit(1)
+
+
+def setup_matmul_subparser(subparsers):
+    matmul_parser = subparsers.add_parser('matmul', help='call with -h for information')
+    matmul_parser.add_argument("--subprogram_main", default=matmul_main, help=argparse.SUPPRESS)
+
+    matmul_parser.add_argument("-b", "--batch-file", default=None, help="batch file used for storing the sample")
+
+    # Interface with benchdnn
+    matmul_parser.add_argument("benchdnn", nargs='?', help="path to benchdnn executable")
+    matmul_parser.add_argument("--engine", default="cpu", help="engine used for benchdnn execution")
+    matmul_parser.add_argument("--collect", default="corr", help="benchdnn collection type, can be one of [corr, perf]")
+    matmul_parser.add_argument("-n", "--name", default="", help="sample name")
+
+    # Sampler Arguments
+    matmul_parser.add_argument("-l", "--layouts", default="all",
+                               help="stag:wtag:dtag, comma separated list of layouts or \"all\" for every supported layout")
+    matmul_parser.add_argument("-m", "--iter-mode", default="zip", help ="iteration mode, must be one of zip or product")
+    matmul_parser.add_argument("-r", "--region", default="(1,1,1,1):(8,8192,8192,8192):(1,1,1,1)",
+                               help="(m_min,k_min,n_min)-(m_max,k_max,n_max)/(m_align,k_align, n_align)")
+    matmul_parser.add_argument("-s", "--samples", default=1000, help="number of samples to collect")
+    matmul_parser.add_argument("-t", "--types", default="all",
+                               help="dt:dt:dt(optional fpmath-mode), comma separated list of type configurations or \"all\" for every supported type")
+def matmul_main(args):
+    batchFile = open(args.batch_file, "w+t") if args.batch_file != None else None
+    if args.benchdnn != None and batchFile == None:
+        batchFile = NamedTemporaryFile("w+t")
+
+    if batchFile:
+        log(f"generating batch file: {batchFile.name}")
+        region = matmul_sampler.Region(args.region)
```

Comment:
The `Region` class is outside of `matmul_sampler`, and needs to be imported/used separately.